### ISSUE TITLE
More robust argument parsing on command line tools

### ISSUE THREAD

**Issue by jessemapel** [DOI-USGS/usgscsm#381]
We can't use the standard getopts header because it's not available on windows. Is there some other argument parsing tool we can use for our scripts? Ideally it will be:

* light-weight. Header only would be best
* Usable for both unix and windows
* Not inhibited by restrictive licensing, think GPL
* Available on Conda

See #380 and #377 for more context

**Comment by oleg-alexandrov** [comment]
I guess the question here is if we need a  command line parser which is general enough to handle any future parsing needs we could ever think of, or if the little parser I added is not good enough for the job it meant for. About the latter, I am not so sure.

The parser I put in is able to handle anything the user throws at it, and either parse the inputs correctly if those are as documented, or error out with a usage message if the user did not invoke the tool as expected. 

Hence I view this issue as something nice to have rather than something required in order for existing software to function as intended.

**Comment by jessemapel** [comment]
I agree this is mostly a nice to have if we want to build out our command line toolset. It's something that is lacking with our CSM packages at the moment, see https://github.com/USGS-Astrogeology/ale/pull/437

I would like to unify the user focused tools into a single location if possible. Right now things are scattered all over the place between ALE, USGSCSM, Knoten, and even ISIS.

**Comment by oleg-alexandrov** [comment]
In my experience the boost program options and Google's GFlags are very robust and freely available. ISIS already depends on Boost, so if in the future we surely need a robust parser, we could go with that, maybe, rather than bringing yet another dependency. 

I found getopts.h to be somewhat awkward to use, and unlike boost, it can't print a help message describing each option, which is a nice functionality both for the user and also for the developer as it keeps each option's explanation where the option is defined. 

> I would like to unify the user focused tools into a single location if possible. Right now things are scattered all over the place between ALE, USGSCSM, Knoten, and even ISIS.

The Python tools need a Python run-time of a specific version installed with them, which I think is a little annoying, but I guess the user would not mind if using conda. So Ross's little tool could be added as officially supported under usgscsm as it creates a CSM model.

If the knoten tools depend on GDAL likely we can't bring them in since then usgscsm would not play nice with isis itself so maybe they are best left where they are.

**Comment by oleg-alexandrov** [comment]
The little usgscsm_cam_test program is about the same as it was 4 years ago. The doc page is quite comprehensive. Robust command-line parsing would be nice, I guess, but adding a new library as dependency is likely not justified given that this is a little useful sanity check tool that is not a primary shippable product. 

Closing this too as likely not controversial.
